scripts/joystick_controls_node

```diff
@@ -0,0 +1,101 @@
+#! /usr/bin/env python
+
+import sys
+
+import rospy
+import rosnode
+from sensor_msgs.msg import Joy
+from geometry_msgs.msg import Twist
+from sensor_msgs.msg import JointState
+from robot_skills.get_robot import get_robot
+from robot_skills.arm import arms
+
+class JoystickControls(object):
+    def __init__(self):
+        """
+        Subscribes to the Joy messages sent via the controller and the joy node and then transforms these to the desired behavior
+        """
+        self.controller_mode = 4
+        self.controller_sub = rospy.Subscriber("joy", Joy, self.map_joystick, queue_size=1)
+        self.gripper_state_sub = rospy.Subscriber("joint_states", JointState, self.get_gripper_state, queue_size=1)
+
+        self.base_pub = rospy.Publisher("base/references", Twist, queue_size=1)
+        self.robot = get_robot('hero')
+        self.robot_arm = self.robot.get_arm(required_gripper_types=[arms.GripperTypes.GRASPING], required_goals=['show_gripper'])
+        self.gripper_state = None
+
+        # set all out-msgs
+        self.base_command = Twist()
+        # self.head_command = None
+
+        # send controls at constant rate
+        r = rospy.Rate(10) # 10hz
+        while not rospy.is_shutdown():
+            self.publish_controls()
+            r.sleep()
+
+    def _send_controls(self, mapping):
+        self.base_command = Twist()
+        # First check if any state machines need killing since this likely takes longest
+        if mapping['B']:
+            rospy.loginfo("Killing existing statemachine.")
+            rosnode.kill_nodes("/state_machine")        # kill whatever statemachine (=challenge/free_mode) is currently running
+        if mapping['A']:
+            if self.gripper_state == "open":
+                self.robot_arm.gripper.send_goal('close')
+            elif self.gripper_state == "closed":
+                self.robot_arm.gripper.send_goal('open')
+        if mapping['X']:
+            self.robot.reset()
+        if mapping['Y']:
+            self.robot_arm.send_joint_goal('show_gripper')
+        if abs(mapping['right_stick_horizontal']) > 0.1 or abs(mapping['right_stick_vertical']) > 0.1: # linear base movement
+            self.base_command.linear.x = 0.4 * mapping['right_stick_vertical']
+            self.base_command.linear.y = 0.4 * mapping['right_stick_horizontal']
+        if abs(mapping['left_stick_horizontal']) > 0.1: # angular base movement
+            self.base_command.angular.z = 0.6 * mapping['left_stick_horizontal']
+        if mapping['right_bumper']:
+            self.robot.speech.speak("Beep. Boop. Beep boop.")
+
+        if abs(mapping['right_stick_horizontal']) < 0.1 and abs(mapping['right_stick_vertical']) < 0.1 and abs(mapping['left_stick_horizontal']) < 0.1:
+            self.base_command = Twist()
+
+    def publish_controls(self):
+            self.base_pub.publish(self.base_command)
```

Comment:
Zit hier geen risico in als er geen input meer komt. Gaat ie dan niet eindeloos doorrijden.